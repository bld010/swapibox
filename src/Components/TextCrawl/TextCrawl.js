import React from 'react';
import PropTypes from 'prop-types';
import './TextCrawl.scss'

const TextCrawl = (props) => {



  return(
    <section className="TextCrawl">
      <div className="crawl">
        <p>{props.movie && props.movie.opening_crawl}</p>
        <p>{props.movie && props.movie.title}</p>
        <p>{props.movie && props.movie.release_date}</p>
      </div>
    </section>
  )
}

export default TextCrawl;

TextCrawl.propTypes = {
  movie: PropTypes.object,
}

