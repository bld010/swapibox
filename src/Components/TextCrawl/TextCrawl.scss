.TextCrawl {
  width: 400px;
  color: goldenrod;
  & p {

    text-justify: inter-character;
  }
}