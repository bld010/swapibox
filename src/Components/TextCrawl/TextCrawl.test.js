import React from 'react';
import { shallow } from 'enzyme';
import TextCrawl from './TextCrawl';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('TextCrawl', () => {
  it('should match the snapshot', () => {

    const mockMovie = {
        opening_crawl: 'A long time ago in a galaxy far far away',
        title:'Star Wars',
        release_date: 'Yesteryear'
      }
    
    const wrapper = shallow(<TextCrawl 
      movie={mockMovie}
    />)

    expect(wrapper).toMatchSnapshot();

  })
})