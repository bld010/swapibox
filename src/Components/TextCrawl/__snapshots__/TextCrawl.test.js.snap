// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextCrawl should match the snapshot 1`] = `
<section
  className="TextCrawl"
>
  <div
    className="crawl"
  >
    <p>
      A long time ago in a galaxy far far away
    </p>
    <p>
      Star Wars
    </p>
    <p>
      Yesteryear
    </p>
  </div>
</section>
`;
